packages/vite/src/node/config.ts

```diff
@@ -330,17 +337,19 @@ export async function resolveConfig(
   configEnv.mode = mode
 
```

Comment:
Adding this [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) for proper type narrowing (resolves `any` in `p: any` in the `asyncFlatten()` call below):

```suggestion
const isPlugin = (plugin: any): plugin is Plugin => typeof plugin.apply !== 'undefined'
```

Reply:
I'm not sure if this is correct. A Vite plugin doesn't need an `apply` prop. I could be undefined so it applies to serve and build

Reply:
Oh, I see now.